clang/test/AST/ast-dump-type-callingconv.cpp

```diff
@@ -0,0 +1,24 @@
+// Verify there is a space after the parens when priting callingconv attributes.
+// RUN: %clang_cc1 -DDEVICE -triple spirv64 -ast-dump -ast-dump-filter foo %s \
+// RUN: | FileCheck -check-prefix=CHECK-DEVICE --strict-whitespace %s
+
+// RUN: %clang_cc1 -DVECTOR -triple aarch64 -ast-dump -ast-dump-filter foo %s \
+// RUN: | FileCheck -check-prefix=CHECK-VECTOR --strict-whitespace %s
+
+// RUN: %clang_cc1 -DSVE -triple aarch64 -ast-dump -ast-dump-filter foo %s \
+// RUN: | FileCheck -check-prefix=CHECK-SVE --strict-whitespace %s
+
+#ifdef DEVICE
+// CHECK-DEVICE-NOT: ()__attribute__((device_kernel))
+void foo() __attribute__((device_kernel));
+#endif
+
+#ifdef VECTOR
+// CHECK-VECTOR-NOT: ()__attribute__((aarch64_vector_pcs))
+void foo()  __attribute__((aarch64_vector_pcs));
+#endif
+
+#ifdef SVE
+// CHECK-SVE-NOT: ()__attribute__((aarch64_sve_pcs))
+void foo()  __attribute__((aarch64_sve_pcs));
+#endif
```

Comment:
Can you make this test a positive pattern, instead of a CHECK-NOT? Otherwise the test is a bit fragile.
This shouldn't be problem, since you already have strict-whitespace passed to FileCheck.

Reply:
Also, with a positive pattern, it shouldn't be necessary to run the test three separate times, and you can remove the `#ifdef`s

Reply:
Thanks for the review. I did the negative check because before this change the AST print is a little different for some reason where it contains the good string as part of the overall string, ex:
```
foo3 'void () __attribute__((aarch64_sve_pcs))':'void ()__attribute__((aarch64_sve_pcs))'
```
and after the fix the output is only
```
foo3 'void () __attribute__((aarch64_sve_pcs))'
```
so checking for `foo3 'void () __attribute__((aarch64_sve_pcs))'` doesn't work because it's a substring in the bad case.


but I managed to fix it by using regex and checking for EOL, it's not super pretty though, just pushed it, thanks.